src/display/editor/tools.js

```diff
@@ -904,42 +939,63 @@ class AnnotationEditorUIManager {
       return;
     }
 
+    this.unselectAll();
+
     const layer = this.#allLayers.get(this.#currentPageIndex);
     const newEditors = this.#clipboardManager
       .paste()
       .map(data => layer.deserialize(data));
```

Comment:
*A similar sort of question as above applies here as well.*

Reply:
The same, we'll undo/redo on the current state and not on what it was before.